### PR TITLE
low-code: Add string filter 

### PR DESCRIPTION
**Pull request by girarda**
## What
Add a function allowing users to convert an input value to a string
```
{{ config['channel_filter'] | string }}
```

Closes
https://github.com/airbytehq/airbyte-internal-issues/issues/6771

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml
Comment (natikgadzhi) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2402]: Nit: other descriptions seem to have the period in the end. 

```suggestion
      description: Converts the specified value to a string.
```

**Review by natikgadzhi — APPROVED** [review]
This looks great, thank you for the quick win! 

Looking at formatting error in CI, I'm not clear what exactly fails — looks like prettier spits out an error from the declarative_component_schema.

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py
Comment (lmossman) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py:94]: @girarda it looks like this change also means that the CDK will no longer complain when someone tries to pass in a non-stringified array value to a query parameter.

For example, when I try this on cloud, I get the error shown:
Screenshot 2024-03-22 at 5 12 00 PM

But when I try this on your branch, it lets me set it:
Screenshot 2024-03-22 at 5 12 14 PM
and if I make it fail by giving it a bogus URL path (so that I can see the actual url-encoded request URL), I can see that it is actually doing the approach where it sets the same key multiple times:
Screenshot 2024-03-22 at 5 14 12 PM

Maybe this is actually what we want? Because then for [this request](https://airbytehq-team.slack.com/archives/C03GD9SV36E/p1711040439412539?thread_ts=1710540234.262429&cid=C03GD9SV36E), the default behavior does what they want (splitting the values across multiple duplicate keys), but if someone doesn't want that behavior, they can do the `| string` macro that you added in this PR to treat the value as a raw string instead.

Comment (girarda) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py:94]: support for passing the array as multiple query params was added yesterday because it was blocking a migration https://github.com/airbytehq/airbyte/pull/36357

Comment (lmossman) [reply airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/filters.py:94]: Ah I understand, thanks for clarifying

**Review by lmossman — APPROVED** [review]

**Comment by lmossman** [comment]
@girarda one thing I forgot to suggest before approving (but doesn't need to block merging this as-is):

We should also update the low-code CDK / builder docs to cover these filters so that people can read about them

**Comment by girarda** [comment]
Good call @lmossman! I think it's only a matter of updating https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/config-based/understanding-the-yaml-file/reference.md.

I'll do that in a separate PR

**Pull request merged**
